framework/src/ble_manager/ble_manager_state.c

```diff
@@ -197,6 +267,137 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
 		ret = ble_drv_conn_is_any_active(is_active);
 	} break;
 
+	// Scanner
+	case BLE_EVT_CMD_START_SCAN: {
+		BLE_STATE_CHECK;
+
+		if (g_scan_ctx.state != BLE_SCAN_STOPPED) {
```

Comment:
So you should consider 2 things.
```
1. It is related to state, there should be lock mechanism.
2. State of framework should aligned with driver. Many of application can use framework, but driver will be used by only framework. So you should add some mechanism to get state of driver.
   For example, If application terminated suddenly and restarted it will init framework again. Then framework's state will be changed as a initial state, but driver still running.
```

Reply:
1. Same as above.  BLE APIs are guaranteed to process in sequence.
2. Yes, you are right. I will raise a new PR for synchronizing state between driver and framework